tests/unit/tests_high_five_base.cpp

```diff
@@ -667,11 +669,48 @@ void readWriteVectorTest() {
     BOOST_CHECK_EQUAL(vec.size(), x_size);
     BOOST_CHECK_EQUAL(result.size(), x_size);
 
+    for (size_t i = 0; i < x_size; ++i) {
+        // std::cout << result[i] << " " << vec[i] << "  ";
+        BOOST_CHECK_EQUAL(result.at(i), vec.at(i));
+    }
+}
+BOOST_AUTO_TEST_CASE_TEMPLATE(readWriteVector, T, numerical_test_types) {
+    readWriteVectorTest<T>();
+}
+
+template <typename T>
+void readWriteArrayTest() {
+    using namespace HighFive;
+
+    std::ostringstream filename;
+    filename << "h5_rw_arr_" << typeid(T).name() << "_test.h5";
+
+    const size_t x_size = 800;
+    const std::string DATASET_NAME("dset");
+    typename std::array<T, x_size> vec;
+    for (size_t i = 0; i < x_size; i++)
+        vec.at(i) = i % 100;
+
+    // Create a new file using the default property lists.
+    File file(filename.str(), File::ReadWrite | File::Create | File::Truncate);
+
+    // Create a dataset with double precision floating points
+    DataSet dataset = file.createDataSet<T>(DATASET_NAME, DataSpace::From(vec));
+
+    dataset.write(vec);
+
+    typename std::array<T, x_size> result;
+
+    dataset.read(result);
```

Comment:
Here I forgot to ask for

    typename std::array<T, 1> tooSmall;
    BOOST_CHECK_THROW(dataset.read(tooSmall), DataSpaceException)

Do you mind creating another PR for it?